roachtest/ttl-restart: make test robust to lease placement variability

spilchen opened the pull request:
The ttl_restart roachtest was flaky due to its reliance on having one lease per node. It attempted to enforce this distribution by relocating leases before starting the TTL job. However, this setup was not always effective, and the resulting imbalance sometimes caused the test to fail.

This change improves the test's resilience by checking the lease distribution after the TTL job has started running. If the TTL job does not have one lease per node at that point, the test logs an explanatory message and exits early, treating the run as successful.

Fixes #151112
Fixes #151113

Release note: None
Epic: none

rafiss reviewed (approved):
nice find. this lgtm, but I wonder if we need to do something to make sure we don't hit the short-circuit success regularly?

spilchen committed:
roachtest/ttl-restart: make test robust to lease placement variability

The ttl_restart roachtest was flaky due to its reliance on having one
lease per node. It attempted to enforce this distribution by relocating
leases before starting the TTL job. However, this setup was not always
effective, and the resulting imbalance sometimes caused the test to
fail.

This change improves the test's resilience by checking the lease
distribution after the TTL job has started running. If the TTL job does
not have one lease per node at that point, the test logs an explanatory
message and exits early, treating the run as successful.

Fixes #151112
Fixes #151113

Release note: None
Epic: none

spilchen reviewed (commented):
Good idea. I added a retry in case the TTL job isn't distributed evenly. So, it should be pretty rare that it hits the short-circuit success.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/151323#-:-OWv4XYF5ckY7g8s9o7t:b-jq2j2)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
